Examples/Algorithms/Generators/ActsExamples/Generators/VertexGenerators.hpp

```diff
@@ -46,4 +46,40 @@ struct GaussianPrimaryVertexPositionGenerator
   }
 };
 
+//
+struct GaussianDisplacedVertexPositionGenerator
+    : public EventGenerator::PrimaryVertexPositionGenerator {
+  Acts::Vector4 stddev = {0.0, 0.0, 0.0, 0.0};
+  Acts::Vector4 mean = {0.0, 0.0, 0.0, 0.0};
+
+  Acts::Vector4 operator()(RandomEngine& rng) const override {
+    double min_value = -2 * M_PI;  // -π
+    double max_value = 2 * M_PI;   // π
+
+    // Create a uniform real distribution object with range [-π, π)
+    std::uniform_real_distribution<> uniform(min_value, max_value);
+
+    // Create a normal distribution object with mean 0.0 and standard
+    // deviation 1.0
+    auto normal = std::normal_distribution<double>(0.0, 1.0);
+    Acts::Vector4 rnd = {normal(rng), uniform(rng), normal(rng), normal(rng)};
+
+    // Compute the cylindrical coordinates with deviations applied
+    Acts::Vector4 cylindrical = mean + rnd.cwiseProduct(stddev);
+
+    // Extract coordinates from the vector
+    double r = cylindrical(0);
+    double phi = cylindrical(1);
+    double z = cylindrical(2);
+    double t = cylindrical(3);
+    double phi_degrees = phi * (180.0 / M_PI);
+
+    // Convert cylindrical coordinates to Cartesian coordinates
+    double x = r * std::cos(phi_degrees);
+    double y = r * std::sin(phi_degrees);
```

Comment:
Both of these expect `phi` in radians, so no need to convert to degrees! I think this might be why the distribution is a bit funky.